### PR TITLE
Pf/pool prize calc

### PR DESCRIPTION
**Pull request by paulfreund94**
Add prize pool calcs, fix time remaining.

Some front-end tweaks, including updating manifest with a pool together image + title description for the safe app.

### CODE REVIEW

File: src/components/Countdown/Countdown.tsx [hunk 1]
```diff
@@ -20,16 +24,35 @@ const Countdown: React.FC<Props> = ({ prizeGivingSecondsRemaining }: Props) => {
       const time = formatDuration(
         intervalToDuration({
           start: new Date(initializedTime),
-          end: new Date(countdownTimer * 1000),
+          end: new Date(countdownTimer),
         }),
         { delimiter: ', ' },
       );
       const split = time.split(', ');
```
Comment (paulfreund94) [inline src/components/Countdown/Countdown.tsx:31]: tweaked this a bit to handle updating of the countdown timer, removed the seconds just to make it easier from a space perspective, and added the case where the countdown is in the past. shout if you'd prefer it done differently :pray:

File: src/providers/exchange.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  tokenExchange: InitalizedExchangeDict;
+};
+
+// EIGHT DECIMALS
+
+const ExchangeContext = createContext<ExchangeType>({
+  ethToUsd: constants.Zero,
+  tokenExchange: {
+    initialized: false,
+    exchangeDict: {},
```
Comment (paulfreund94) [inline src/providers/exchange.tsx:38]: pull in eth to usd from chainlink contracts, other tokens to eth from uniswap graph

File: src/providers/ethers.tsx [hunk 3]
```diff
@@ -1,19 +1,69 @@
...
+import React, { createContext, useEffect, useContext, useState } from 'react';
 import { SafeAppsSdkProvider } from '@gnosis.pm/safe-apps-ethers-provider';
 import { useSafeAppsSDK } from '@gnosis.pm/safe-apps-react-sdk';
+import { fetch } from 'cross-fetch';
+import { setPtClient, setLbClient } from '../ptGraphClient';
+import { HttpLink, InMemoryCache, ApolloClient } from '@apollo/client';
 
 const EthersContext = createContext<SafeAppsSdkProvider | undefined>(undefined);
 
+const PT_GRAPH_ENDPOINTS: any = {
```
Comment (paulfreund94) [inline src/providers/ethers.tsx:11]: if you open the app in a safe connected to the mainnet  it should now pull in from the mainnet graph, and if you open the app in rinkeby it'll connect to the rinkeby graph.

File: src/utils/calculateEstimatedPrizePool.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+type CalculateEstimatedPoolPrizeArgs = {
+  tokenDecimals: number;
+  awardBalance: BigNumber;
+  poolTotalSupply: BigNumber;
+  supplyRatePerBlock: BigNumber;
+  prizePeriodRemainingSeconds: BigNumber;
+};
+
+export function calculateEstimatedPoolPrize({
+  tokenDecimals,
```
Comment (paulfreund94) [inline src/utils/calculateEstimatedPrizePool.ts:21]: got this from here: https://github.com/pooltogether/pooltogether-reference-pool-ui/blob/master/lib/utils/calculateEstimatedPoolPrize.js

File: src/utils/normalizeTo18Decimals.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { BigNumber } from 'ethers';
+
+export const DEFAULT_TOKEN_DECIMAL_PRECISION = 18;
+
```
Comment (paulfreund94) [inline src/utils/normalizeTo18Decimals.ts:4]: got his from here: https://github.com/pooltogether/pooltogether-reference-pool-ui/blob/master/lib/utils/normalizeTo18Decimals.js

File: src/providers/prize-values.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+          prizeVals.push(truncate(ethers.utils.formatEther(prizeForPool)));
+
+          // ^ just following PT front-end which seems to do this for rinkeby
+          continue;
+        }
+        let poolPrizeEth = derivedEthToEthAlreadyNormalized(
+          prizeForPool,
+          exchangeDict[pools[i].underlyingCollateralContract.address.toLowerCase()].derivedEth,
+        );
+        for (const balance of poolsErc20Balance[i]) {
```
Comment (paulfreund94) [inline src/providers/prize-values.tsx:53]: this is basically adding the value of the lootbox prizes and external erc20 prizes (which are separate I think... lootbox is an erc721 external prize which you can attack erc20 tokens to).

File: src/providers/exchange.tsx (hunk 2 above)
Comment (paulfreund94) [reply src/providers/exchange.tsx:38]: think pool together might be using a different eth to usd oracle which would explain why the prize values we get are slightly different.

File: public/index.html [hunk 7]
```diff
@@ -24,7 +21,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>Gnosis Safe App</title>
+    <title>Pool Together Safe App</title>
```
Comment (DenhamPreen) [inline public/index.html:24]: Yeah boooi

File: src/components/Countdown/Countdown.tsx (hunk 1 above)
Comment (DenhamPreen) [reply src/components/Countdown/Countdown.tsx:31]: No this is totally cool 👌

File: src/pages/Pools/index.tsx [hunk 8]
```diff
@@ -1,18 +1,26 @@
...
   const { ticketBalances, refreshTicketBalances } = useTicketBalances();
   const poolChainData = usePoolChainData();
+  const prizeVals = usePrizeValues();
   useDidMount(() => refreshTicketBalances());
 
   useEffect(() => {
+    console.log('HEY!');
+    console.log(poolChainData);
+    console.log(poolChainData.map((x) => x.secondsRemaining.toNumber()));
+    console.log(Date.now());
```
Comment (DenhamPreen) [inline src/pages/Pools/index.tsx:23]:
```suggestion

```

File: src/providers/exchange.tsx (hunk 2 above)
Comment (DenhamPreen) [reply src/providers/exchange.tsx:38]: This is great, I would say it's probably fine to use off chain api's etc for this kind of data as the chainlink oracle only updates periodically or if there is a 0.5% movement. 
This is 'always free' though so great approach!
https://data.chain.link/eth-usd

Comment (DenhamPreen) [reply src/providers/exchange.tsx:38]: In the future if its easier*, otherwise this is fantastic!!

**Review by DenhamPreen — APPROVED** [review]
This is absolute f***ing heat!!! 
Approved, still need to run it locally just given the code a once over.

**Pull request merged**